lib/Dialect/FIRRTL/Transforms/LowerAnnotations.cpp

```diff
@@ -281,6 +281,31 @@ static LogicalResult drop(const AnnoPathValue &target, DictionaryAttr anno,
 // Customized Appliers
 //===----------------------------------------------------------------------===//
 
+static LogicalResult applyDUTAnno(const AnnoPathValue &target,
+                                  DictionaryAttr anno, ApplyState &state) {
+  auto *op = target.ref.getOp();
+  auto loc = op->getLoc();
+
+  auto opTarget = target.ref.dyn_cast<OpAnnoTarget>();
+  if (!opTarget)
+    return mlir::emitError(loc) << "must target a module object";
```

Comment:
nit: consider combining this "does it target an operation" check with "and is that operation an FModuleOp" together, the error messages are close but different (first and last in this file), sounds like the idea is to complain if targets anything other than an FModuleOp (port, or anything else).
Possibly adding a check for the error if not targeting a Module, but :shrug: .

Would also give early-exit for the failure case, reducing indentation on the non-error path, FWIW.